source/gui/installerGui.py

```diff
@@ -110,17 +110,24 @@ def doInstall(
 			# Translators: The title of a dialog presented to indicate a successful operation.
 			_("Success"))
 	if startAfterInstall:
-		# #4475: ensure that the first window of the new process is not hidden by providing SW_SHOWNORMAL  
-		shellapi.ShellExecute(
-			None,
-			None,
-			os.path.join(installer.defaultInstallPath,'nvda.exe'),
-			None,
-			None,
-			winUser.SW_SHOWNORMAL
-		)
-	else:
-		core.triggerNVDAExit()
+		core.preNVDAExit.register(_startNewInstalledNVDAInstance)

```

Comment:
As preNVDAExit is actually fired at the very top of core.triggerNVDAExit (the line after this) I don't really get how this change makes a difference to the order or timing of execution in order to address the issue... Can you explain further?

Reply:
Currently, `WM_QUIT` exits the mainloop and calls `triggerNVDAExit`

`preNVDAExit` fires the following events in order:

1.  handleNVDAModuleCleanupBeforeGUIExit, which terminates modules
2. _closeAllWindows, which closes all windows and exits the wx mainloop
3. (as of this PR) triggers a new NVDA instance. which should be safe now that all the windows have been closed,  and the wx main loop has exited.

Reply:
Yep, now reading all the changes along with your explanation it makes total sense to me. I now see that  preNVDAExit will have three callbacks registered, in order: 1. handlers, 2. closing windows, and if completing an install, 3. the starting a new NvDA instance.

Reply:
Since the order is important here, please verify that there are unit tests covering the order of callback for extension points.

Reply:
this is covered by `test_registerMultiple` in `tests\unit\test_extensionPoints.py`